### ISSUE TITLE
Allow select as alias custom field name and exclude some fields in prisma client

### ISSUE THREAD

**Issue by ridhwaans** [prisma/prisma#8151]
## Problem

In prisma client 2.26 and sqlite, its not every efficient to run map, reduce server side functions to rename keys in data from prisma. 

If prisma is getting upwards of five different relations, I would rather directly setup aliases, custom field names directly in the query than external code   

similarly, if a table has a lot of field names, theres no clear way to exclude few while selecting all others.   

I dont want to explicitly name every field in the prisma query if I need to select only 24 out of 25 fields in the model relation  

## Suggested solution

I am looking for the prisma equivalent of 
```
SELECT party_id as id FROM genres 
```
and 
```
SELECT * EXCEPT id
FROM (...) genres
```
and somehow apply it across a multiple findManys
starting example
```
var tv_shows = await prisma.tVShow.findMany({
    include: { genres: true, production_companies: true, seasons: { include: { episodes: true } }, credits: true, similar: true }
  })
```

## Alternatives

Doing an external code implementation using `map`, `reduce` to transform the selected object

## Additional context

related thread https://github.com/prisma/prisma/issues/4433

**Comment by dkantereivin** [comment]
I agree this would be a very useful feature. One other option is when selecting fields, to alternatively have the option to simply provide `field: false`, which would deselect them. Many ORM's support a default fetch or not to fetch behaviour (i.e. Mikro's `lazy: true` option, but since Prisma's design is very much data-oriented, I think that would be an antipattern.

**Comment by paulintrognon** [comment]
I don't know if it is the same issue, but I would love to be able to alias (rename) relations as well.

For example, what I would love:
```ts
const customer = await prisma.customer.findUnique({
  where { id },
  select: {
    fruits: {
      select: { id: true },
      where: { type: 'banana' },
      as: 'bananas'
    }
  }
})

// Bananas!
customer.bananas
```

**Comment by philippetedajo** [comment]
Please, Add an alias feature to Prisma

**Comment by benupham** [comment]
Would be nice to have...

**Comment by eliliam** [comment]
Totally seconded, this would make quite a few queries far simpler. I know it presents a few questions and issues around Typescript typings that still need to be resolved, but this would enable some pretty awesome flexibility.

**Comment by j-Kode** [comment]
man I was just searching for this functionality and came here 😢 seems like ill have to created some sort of mapping functionality for now.

**Comment by palashCItobuz** [comment]
I second this as a very good to have feature.

**Comment by shadeemerhi** [comment]
Would also love this feature!

**Comment by CeloHill** [comment]
I would love this feature! Please do it!

**Comment by Gurushanth1** [comment]
I would love this feature! Please do it!

**Comment by SuleymanQ** [comment]
+ 1

**Comment by supryantowp** [comment]
+1

**Comment by stijnvanderlaan** [comment]
> I don't know if it is the same issue, but I would love to be able to alias (rename) relations as well.
> 
> For example, what I would love:
> 
> ```ts
> const customer = await prisma.customer.findUnique({
>   where { id },
>   select: {
>     fruits: {
>       select: { id: true },
>       where: { type: 'banana' },
>       as: 'bananas'
>     }
>   }
> })
> 
> // Bananas !
> customer.bananas
> ```

I would do this a bit differently because it's now not possible to select a different type of fruit, I would do something like this:

```ts

const customer = await prisma.customer.findUnique({
  where { id },
  select: {
   bananas: {
      select: { id: true },
      where: { type: 'banana' },
      alias: 'fruits'
    },
   appels: {
      select: { id: true },
      where: { type: 'appels' },
      alias: 'fruits'
    }
  }
})

// Bananas !
customer.bananas

// Apples !
customer.apples
```

**Comment by kw-michal** [comment]
I guess such feature is not easy to build, as currently all the TS types are generated during client generation phase which is ahead of time. That feature would break the existing types.

**Comment by LeulAria** [comment]
was looking some think like

```ts
          rating_like_dislike: {
            select: {
              liked_by_id: true,
            },
            where: { liked: true },
            as: 'likes'
          },
          rating_like_dislike: {
            select: {
              liked_by_id: true,
            },
            where: { liked: false },
            as: 'unlikes'
          }
```

how can i achive this

**Comment by alocvm** [comment]
Hi there!
What's new about this feature?

**Comment by Onihani** [comment]
+1

**Comment by SwapnilSoni1999** [comment]
i think forever to take?

**Comment by maikonalexandre** [comment]
Já temos esse recurso ? akakkakkd

**Comment by janpio** [comment]
To close the loop here to some related issues:

For the "alias" feature request:
- https://github.com/prisma/prisma/issues/6725
- https://github.com/prisma/prisma/issues/4433

For the "exclude" feature:
- https://github.com/prisma/prisma/issues/5042

**Comment by RajanTpss** [comment]
Hello Folks, I hope you guys are doing well.

Have this issue is being resolved?

Please let me know, I am also facing this same problem.

Thanks

**Comment by ivanvpan** [comment]
@kw-michal hit the nail on the head - this is very difficult with how Prisma works, which is by generating the types prior. Prisma is good for certain things but I would highly recommend everybody to consider a less abstracted DB layer for larger projects. I work for a startup and we outgrew Prisma in a few months.

**Comment by fzn0x** [comment]
Really needed this feature, I'm facing this problem and can't even resolve except using `queryRaw`.

**Comment by pavlyutkin** [comment]
+1

**Comment by faheemhameed** [comment]
I need this pls.

**Comment by LeulAria** [comment]
> I need this pls.

Change orm before its too late 😂😂

**Comment by pavlyutkin** [comment]
> > I need this pls.
> 
> Change orm before its too late 😂😂

It hurts me too after typeorm 😀

**Comment by ghyath5** [comment]
It is a very important feature !!!

It would also be like this 😀: 

```
const customer = await prisma.customer.findUnique({
  where { id },
  select: {
  fruits: [
   {
      select: { id: true },
      where: { type: 'banana' },
      as: 'bananas'
    },
    {
      select: { id: true },
      where: { type: 'appels' },
      alias: 'appels'
    }
    ],
  }
})
```

**Comment by lixaotec** [comment]
How to make 2 includes on the same relation? Is that this issue subject?

**Comment by mrowles** [comment]
Would like this too, yep

**Comment by anasouardini** [comment]
Still no solution to this as well?

**Comment by yassinrais** [comment]
😄

**Comment by ielmar** [comment]
I also have a case where I need to use this.

**Comment by aryan-innovibe** [comment]
Do we not have this yet??

**Comment by ceri-waterscreative** [comment]
+1

**Comment by GithubSPerez** [comment]
XD
